etcdserver/snapshot_store.go

```diff
@@ -25,19 +25,21 @@ import (
 	"github.com/coreos/etcd/raft"
 	"github.com/coreos/etcd/raft/raftpb"
 	dstorage "github.com/coreos/etcd/storage"
+	"github.com/coreos/etcd/storage/backend"
 )
 
 type snapshot struct {
-	r raftpb.Snapshot
+	r  raftpb.Snapshot
+	kv backend.Snapshot
```

Comment:
we should not expose backend to etcdserver. can we define the snapshot interface in kv pkg?